### DISCUSSION TITLE
Support of covariates for Auto Models

### DISCUSSION THREAD

**Discussion by JustM57** [Nixtla/neuralforecast#487] [Q&A] [answered]
Does current implementation of Auto models support covariates? Or will it be in the nearest future?

**Comment by kdgutier** [comment, marked as answer]
Hey @JustM57,

Thanks for using the library.  You can explore exogenous in the `AutoModel` config using a list of lists like this:
```hist_exog_list = [[‘x1’],[‘x1’, ‘x2’]]```

**Reply by JustM57** [reply]
List of lists? So it's kind of when ray chooses the best subset of covariates? And if the set is fixed the length of outer list is 1.

**Reply by kdgutier** [reply]
Yes, ray uses `hyperopt` in the background to automatically explore the validation signal and select from the list the best performing set of covariates. This covariate selection technique is good as long as your subset exploration does not grow so fast.
